x-pack/plugins/apm/server/lib/helpers/spans/get_is_using_service_destination_metrics.ts

```diff
@@ -67,28 +68,51 @@ export async function getIsUsingServiceDestinationMetrics({
 }) {
   const { apmEventClient } = setup;
 
-  const response = await apmEventClient.search(
-    'get_has_service_destination_metrics',
-    {
-      apm: {
-        events: [getProcessorEventForServiceDestinationStatistics(true)],
-      },
-      body: {
-        terminate_after: 1,
-        size: 1,
-        query: {
-          bool: {
-            filter: [
-              ...rangeQuery(start, end),
-              ...kqlQuery(kuery),
-              ...getDocumentTypeFilterForServiceDestinationStatistics(true),
-              ...(useSpanName ? [{ exists: { field: SPAN_NAME } }] : []),
-            ],
+  async function getServiceDestinationMetricsCount(
+    query?: QueryDslQueryContainer
+  ) {
+    const response = await apmEventClient.search(
+      'get_service_destination_metrics_count',
+      {
+        apm: {
+          events: [getProcessorEventForServiceDestinationStatistics(true)],
+        },
+        body: {
+          size: 1,
+          terminate_after: 1,
+          query: {
+            bool: {
+              filter: [
+                ...rangeQuery(start, end),
+                ...kqlQuery(kuery),
+                ...getDocumentTypeFilterForServiceDestinationStatistics(true),
+                ...(query ? [query] : []),
+              ],
+            },
           },
         },
-      },
-    }
-  );
+      }
+    );
+
+    return response.hits.total.value;
+  }
 
-  return response.hits.total.value > 0;
+  if (!useSpanName) {
+    return (await getServiceDestinationMetricsCount()) > 0;
+  }
```

Comment:
The way `useSpanName` is used is still pretty fuzzy to me. It's set in a context (`TimeRangeMetadataContextProvider`) based on the selected client side route. The behaviour feels a little too in-directional and I worry that someone in 6 months will change this and accidentally break it.

Reply:
The difficulty is that `TimeRangeMetadataContextProvider` takes care of the data fetching. I can't really think of a reliable way to configure that from outside of the provider, e.g. in the consumer of the context, as the context is to be shared across the application. Do you have a suggestion?

Reply:
I don't have a suggestion right now. Perhaps we can think of another approach down the line.